styx-flyte-client/src/main/java/com/spotify/styx/flyte/client/FlyteAdminClient.java

```diff
@@ -0,0 +1,91 @@
+/*-
+ * -\-\-
+ * Spotify Styx Flyte Client
+ * --
+ * Copyright (C) 2016 - 2020 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.flyte.client;
+
+import static com.google.common.base.Verify.verifyNotNull;
+
+import flyteidl.admin.ExecutionOuterClass;
+import flyteidl.service.AdminServiceGrpc;
+import io.grpc.ManagedChannelBuilder;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FlyteAdminClient {
+
+  private static final Logger log = LoggerFactory.getLogger(FlyteAdminClient.class);
+  private static final String TRIGGERING_PRINCIPAL = "styx";
+  private static final int USER_TRIGGERED_EXECUTION_NESTING = 0;
+
+  private final AdminServiceGrpc.AdminServiceBlockingStub stub;
+
+  FlyteAdminClient(AdminServiceGrpc.AdminServiceBlockingStub stub) {
```

Comment:
Let's annotate this with `visible for test`.

Reply:
👍